community/modules/scheduler/htcondor-central-manager/main.tf

```diff
@@ -0,0 +1,169 @@
+ */
+
+locals {
+  # This label allows for billing report tracking based on module.
+  labels = merge(var.labels, { ghpc_module = "htcondor-central-manager" })
+}
+
+locals {
+  network_storage_metadata = var.network_storage == null ? {} : { network_storage = jsonencode(var.network_storage) }
+  oslogin_api_values = {
+    "DISABLE" = "FALSE"
+    "ENABLE"  = "TRUE"
+  }
+  enable_oslogin_metadata = var.enable_oslogin == "INHERIT" ? {} : { enable-oslogin = lookup(local.oslogin_api_values, var.enable_oslogin, "") }
+  metadata                = merge(local.network_storage_metadata, local.enable_oslogin_metadata, var.metadata)
+
+  name_prefix = "${var.deployment_name}-cm"
+
+  all_runners = flatten([var.central_manager_runner, local.schedd_runner])
+
+  cm_config = templatefile("${path.module}/templates/condor_config.tftpl", {})
+
+  cm_object = "gs://${var.htcondor_bucket_name}/${google_storage_bucket_object.cm_config.output_name}"
+  schedd_runner = {
+    type        = "ansible-local"
+    content     = file("${path.module}/files/htcondor_configure.yml")
+    destination = "htcondor_configure.yml"
+    args = join(" ", [
+      "-e config_object=${local.cm_object}",
+    ])
+  }
+
+  central_manager_ips  = [data.google_compute_instance.cm.network_interface[0].network_ip]
+  central_manager_name = data.google_compute_instance.cm.name
+
+  list_instances_command = "gcloud compute instance-groups list-instances ${data.google_compute_region_instance_group.cm.name} --region ${var.region} --project ${var.project_id}"
+}
+
+data "google_compute_image" "htcondor" {
+  family  = var.instance_image.family
+  project = var.instance_image.project
+
+  lifecycle {
+    postcondition {
+      condition     = self.disk_size_gb <= var.disk_size_gb
+      error_message = "var.disk_size_gb must be set to at least the size of the image (${self.disk_size_gb})"
+    }
+  }
+}
+
+data "google_compute_zones" "available" {
+  project = var.project_id
+  region  = var.region
+}
+
+data "google_compute_region_instance_group" "cm" {
+  self_link = time_sleep.mig_warmup.triggers.self_link
+}
+
+data "google_compute_instance" "cm" {
+  self_link = data.google_compute_region_instance_group.cm.instances[0].instance
+}
+
+resource "google_storage_bucket_object" "cm_config" {
+  name    = "${var.deployment_name}-cm-config-${substr(md5(local.cm_config), 0, 4)}"
+  content = local.cm_config
+  bucket  = var.htcondor_bucket_name
+}
+
+module "startup_script" {
+  source = "github.com/GoogleCloudPlatform/hpc-toolkit//modules/scripts/startup-script?ref=v1.20.0&depth=1"
+
+  project_id      = var.project_id
+  region          = var.region
+  labels          = local.labels
+  deployment_name = var.deployment_name
+
+  runners = local.all_runners
+}
+
+module "central_manager_instance_template" {
+  # tflint-ignore: terraform_module_pinned_source
+  source = "github.com/terraform-google-modules/terraform-google-vm//modules/instance_template?ref=84d7959"
```

Comment:
nit: I think I would prefer to use a version ref. Any reason to select a specific commit?